caffe2/operators/mean_op.h

```diff
@@ -72,6 +114,34 @@ class MeanOp final : public Operator<Context> {
           Input(0).dtype().name());
     }
   }
+
+ private:
+  std::vector<int64_t> ComputeBroadcastDims(
+      const std::vector<std::vector<int>>& input_dims) const {
+    const int ninp = input_dims.size();
+    std::vector<int> input_sizes(ninp);
+    for (int i = 0; i < ninp; i++) {
+      input_sizes[i] = input_dims[i].size() - 1;
+    }
+
+    const int ndim = *std::max_element(input_sizes.begin(), input_sizes.end());
+    std::vector<int64_t> output_dims(ndim + 1);
+    for (int k = ndim; k >= 0; k--) {
+      int max_dim = -1;
+      for (int i = 0; i < ninp; i++) {
+        int isz = input_sizes[i];
+        if (isz >= 0) {
+          CAFFE_ENFORCE(
+              max_dim == -1 || max_dim == 1 || input_dims[i][isz] == max_dim ||
```

Comment:
Actually, there is one weird behavior is if a's shape is [1, 1] and b's shape is [0], mean(a, b) should have shape [1, 0]. I know this is weird, but if we are following the numpy's broadcasting, we should do so. I am not sure how Add/Sub/... is implemented. I think we can fix this later. cc: @BIT-silence